src/main/scala/org/hammerlab/guacamole/commands/GermlineAssemblyCaller.scala

```diff
@@ -59,6 +59,57 @@ object GermlineAssemblyCaller {
     override val description = "call germline variants by assembling the surrounding region of reads"
 
     /**
+     * Extract a subsequence from a read sequence
+     *
+     * NOTE: This assumes that the read entirely covers the reference region and does not check for insertions
+     * or deletion
+     *
+     * @param read Read to extract subsequence from
+     * @param startLocus Start (inclusive) locus on the reference
+     * @param endLocus End (exclusive) locus on the reference
+     * @return Subsequence overlapping [startLocus, endLocus)
+     */
+    def getSequenceFromRead(read: MappedRead, startLocus: Int, endLocus: Int) = {
+      read.sequence.slice(startLocus - read.unclippedStart.toInt, endLocus - read.unclippedStart.toInt)
+    }
+
+    /**
+     * For a given set of reads identify all kmers that appear in the specified reference region
+     *
+     * @param reads  Set of reads to extract sequence from
+     * @param startLocus Start (inclusive) locus on the reference
+     * @param endLocus End (exclusive) locus on the reference
+     * @param kmerSize Length of the subsequence
+     * @param minOccurrence Mininum number of times a subsequence needs to appear to be included
+     * @return List of subsequences overlapping [startLocus, endLocus) that appear at least `minOccurrence` time
+     */
+    def getConsensusKmer(reads: Seq[MappedRead],
+                         startLocus: Int,
+                         endLocus: Int,
+                         kmerSize: Int,
+                         minOccurrence: Int) = {
+
+      // Filter to reads that entirely cover the region
+      // Exclude reads that have any non-M Cigars (these don't have a 1 to 1 base mapping to the region)
+      val overlapping = reads
+        .filter(r => r.cigarElements.count(el => el.getOperator != CigarOperator.M) == 0)
```

Comment:
Updated to the `exists`. Yea, for now I'm filtering out anything non-M since otherwise it's difficult to figure out where that reference base is in the read sequence without processing the full cigar. i.e. if their was a insertion/deletion earlier in the sequence the position would need to be shifted